com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/PollServiceInProgessStateExecution.java

```diff
@@ -0,0 +1,106 @@
+package com.sap.cloud.lm.sl.cf.process.steps;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import org.apache.commons.collections4.MapUtils;
+import org.cloudfoundry.client.lib.CloudControllerClient;
+import org.cloudfoundry.client.lib.domain.CloudEvent;
+import org.flowable.engine.delegate.DelegateExecution;
+
+import com.sap.cloud.lm.sl.cf.client.lib.domain.CloudServiceExtended;
+import com.sap.cloud.lm.sl.cf.core.cf.clients.EventsGetter;
+import com.sap.cloud.lm.sl.cf.core.cf.clients.ServiceGetter;
+import com.sap.cloud.lm.sl.cf.core.cf.services.ServiceOperation;
+import com.sap.cloud.lm.sl.cf.core.cf.services.ServiceOperationState;
+import com.sap.cloud.lm.sl.cf.core.cf.services.ServiceOperationType;
+import com.sap.cloud.lm.sl.cf.process.util.ServiceOperationExecutor;
+
+public class PollServiceInProgessStateExecution extends PollServiceOperationsExecution implements AsyncExecution {
+
+    private static final String LAST_SERVICE_OPERATION = "last_operation";
+    private static final String SERVICE_OPERATION_TYPE = "type";
+    private static final String SERVICE_OPERATION_STATE = "state";
+    private static final String SERVICE_OPERATION_DESCRIPTION = "description";
+
+    private ServiceOperationExecutor serviceOperationExecutor = new ServiceOperationExecutor();
+    private ServiceGetter serviceInstanceGetter;
+    private EventsGetter eventsGetter;
+
+    public PollServiceInProgessStateExecution(ServiceGetter serviceInstanceGetter, EventsGetter eventsGetter) {
+        this.serviceInstanceGetter = serviceInstanceGetter;
+        this.eventsGetter = eventsGetter;
+    }
+
+    @Override
+    protected List<CloudServiceExtended> computeServicesToPoll(ExecutionWrapper execution,
+        Map<String, ServiceOperationType> triggeredServiceOperations) {
+        List<CloudServiceExtended> existingServices = getServices(execution.getContext());
+        return getServicesWithTriggeredOperations(existingServices, triggeredServiceOperations);
+    }
+
+    private List<CloudServiceExtended> getServices(DelegateExecution context) {
+        Set<Entry<String, CloudServiceExtended>> allServicesToPoll = StepsUtil.getServicesData(context)
+            .entrySet();
+        // There's no need to poll the user-provided services, because it is done synchronously:
+        return allServicesToPoll.stream()
+            .filter(service -> !service.getValue()
+                .isUserProvided())
+            .map(Map.Entry::getValue)
```

Comment:
Here you only need the CloudServiceExtended, why passing the map(applies to my above comment from the Step).